content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "CI/CD Pipelines for Kubernetes Apps with Pulumi & Codefresh"
+date: 2021-01-19
+meta_desc: "Build Kubernetes applications and provision infrastructure with Pulumi and Codefresh's CI/CD pipelines"
+meta_image: codefresh.png
+authors:
+    - sophia-parafina
+tags:
+    - CI/CD
+    - Codefresh
+    - Kubernetes
+
+---
+
+Delivering modern applications is complicated and requires coordination of many moving parts. Applications are frequently updated to implement new features and improve security and performance which translates to a better user experience for your customers. To further complicate matters, infrastructure must also be deployed and maintained simultaneously with applications to avoid conflicts or dependencies.
+
+Containerized applications deployed on Kubernetes are particularly susceptible to a misalignment between developers who frequently push changes and operators who want to maintain a stable architecture. Continuous Integration builds and tests software and delivers it as packages. Continuous Delivery or Deployment deploys applications on infrastructure. Let's take a look at how we can accomplish CI/CD for both applications and infrastructure.
+
+<!--more-->
+
+## CI/CD and Pipelines
+
+CD is sometimes called Continuous Delivery or Continuous Deployment. These have the same results, i.e., deploying software artifacts to production. However, Continuous Deployment is an automated process that deploys artifacts without human intervention. Continuous Delivery occurs when a person decides when to deploy an application. The reason could be a business decision, such as coordinating the release with a marketing campaign or having a final check by an engineer before release.
+
+Continuous integration is an automated process that begins with a code push to a repository. From there, the code can be built and tested and eventually released as an artifact. The CI process is determined by your organization and it can include manual testing of the code to catch corner cases.
+
+A pipeline is the deployable unit path for CI/CD. A pipeline starts when code is committed to a repository like GitHub. Next step is a notification to a build system, such as Codefresh. The build system compiles the code and runs unit tests. If your code passes the unit tests, integration tests are the next step. In the case of containers, if your code passes both unit and integration tests, the images will be created and pushed into a registry service. This is the simplest example of a pipeline, but you can do many more things such as security scans, check modules for CVEs (Common Vulnerabilities and Exploits), send Slack notifications, and run quality checks. A pipeline can be fully automated or have checkpoints that require approval before resuming.
+
+< ADD CODEFRESH CONTENT HERE >
```

Comment:
Not sure if @kostis-codefresh was planning to add anything here.

```suggestion
```

Reply:
YesI have some content. @spara I have updated our google doc. Please see changes there.

Reply:
added